src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -164,6 +164,13 @@ public static FormulaValue FilterTable(EvalVisitor runner, SymbolContext symbolC
             return new InMemoryTableValue(irContext, rows);
         }
 
+        public static FormulaValue IndexTable(IRContext irContext, FormulaValue[] args)
+        {
+            var arg0 = (TableValue)args[0];
+            var arg1 = (NumberValue)args[1];
+            return arg0.Rows.ElementAt((int)arg1.Value).ToFormulaValue();
```

Comment:
Check range first and return an error if the index is out of bounds

Reply:
added range check